<a href="https://colab.research.google.com/github/nurahmi-tintyasusi-dev/data-science-2026/blob/main/Pertemuan6_Nurahmi_Tintyasusi_250401020035.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Nama : NURAHMI TINTYASUSI

NIM : 250401020035

Kelas : IF401

Mata Kuliah : Data Science


---




# **Load dan Eksplorasi Dataset**

In [27]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

df_titanic = sns.load_dataset('titanic')

#pilih Kolom yang akan digunakan
kolom_pilihan = ['pclass', 'sex', 'age', 'sibsp', 'parch', 'fare', 'embarked', 'survived']
df_titanic = df_titanic[kolom_pilihan].copy()

print('Shape:', df_titanic.shape)
print('\nMissing values:')
print(df_titanic.isnull().sum())
print('\nDistribusi target:')
print(df_titanic['survived'].value_counts(normalize=True).round(3))
df_titanic.head()


Shape: (891, 8)

Missing values:
pclass        0
sex           0
age         177
sibsp         0
parch         0
fare          0
embarked      2
survived      0
dtype: int64

Distribusi target:
survived
0    0.616
1    0.384
Name: proportion, dtype: float64


,pclass,sex,age,sibsp,parch,fare,embarked,survived
0,3,male,22.0,1,0,7.2500,S,0
1,1,female,38.0,1,0,71.2833,C,1
2,3,female,26.0,0,0,7.9250,S,1
3,1,female,35.0,1,0,53.1000,S,1
4,3,male,35.0,0,0,8.0500,S,0



# Imputasi missing values

In [28]:
df_titanic['age'] = df_titanic['age'].fillna(df_titanic['age'].median())
df_titanic['embarked'] = df_titanic['embarked'].fillna(df_titanic['embarked'].mode()[0])

print("Missing setelah imputasi:")
print(df_titanic.isnull().sum())



Missing setelah imputasi:
pclass      0
sex         0
age         0
sibsp       0
parch       0
fare        0
embarked    0
survived    0
dtype: int64



# Encoding Kategorikal

In [29]:
df_titanic = pd.get_dummies(df_titanic, columns=['sex', 'embarked'], drop_first=True, dtype=int)

print("Kolom setelah encoding:")
print(df_titanic.columns.tolist())


Kolom setelah encoding:
['pclass', 'age', 'sibsp', 'parch', 'fare', 'survived', 'sex_male', 'embarked_Q', 'embarked_S']


# Train-Test Split

In [30]:
from sklearn.model_selection import train_test_split

X = df_titanic.drop('survived', axis=1)
y = df_titanic['survived']

X_train, X_test, y_train, y_test = train_test_split( X, y,
                                                    test_size=0.2, random_state=42, stratify=y )

print(f"Train: {X_train.shape[0]} baris")
print(f"Test: {X_test.shape[0]} baris")

proporsi_train = y_train.value_counts(normalize=True).round(3)
proporsi_test = y_test.value_counts(normalize=True).round(3)

print("\nProporsi survived di Train:")
print(proporsi_train)
print("\nProporsi survived di Test:")
print(proporsi_test)

Train: 712 baris
Test: 179 baris

Proporsi survived di Train:
survived
0    0.617
1    0.383
Name: proportion, dtype: float64

Proporsi survived di Test:
survived
0    0.615
1    0.385
Name: proportion, dtype: float64


# Feature Scalling

In [31]:


from sklearn.preprocessing import StandardScaler

kolom_numerik = ['pclass', 'age', 'sibsp', 'parch', 'fare']
scaler = StandardScaler()

X_train[kolom_numerik] = scaler.fit_transform(X_train[kolom_numerik])
X_test[kolom_numerik] = scaler.transform(X_test[kolom_numerik])

print("Mean dari training:", scaler.mean_.round(2))
print("Std dari training:", scaler.scale_.round(2))

print("\nContoh X_train setelah scaling:")
print(X_train.head(3).round(3))

print(f"\nData siap untuk model Machine Learning!")
print(f"X_train: {X_train.shape}, y_train: {y_train.shape}")
print(f"X_test: {X_test.shape}, y_test: {y_test.shape}")


Mean dari training: [ 2.31 29.46  0.49  0.39 31.82]
Std dari training: [ 0.83 13.03  1.06  0.84 48.03]

Contoh X_train setelah scaling:
     pclass    age  sibsp  parch   fare  sex_male  embarked_Q  embarked_S
692   0.830 -0.112 -0.465 -0.466  0.514         1           0           1
481  -0.371 -0.112 -0.465 -0.466 -0.663         1           0           1
527  -1.571 -0.112 -0.465 -0.466  3.955         1           0           1

Data siap untuk model Machine Learning!
X_train: (712, 8), y_train: (712,)
X_test: (179, 8), y_test: (179,)


# KESIMPULAN

*  **Apa yang dipelajari:** Mempelajari dan mempraktikkan alur (pipeline) data preprocessing yang benar sebelum data siap dimasukkan ke dalam model Machine Learning
*  **Temuan Utama:** Setelah dilakukan stratified split, data terbagi menjadi 712 baris untuk training dan 179 baris untuk testing dengan proporsi kelas target yang terjaga konsisten di angka ~61% (tidak selamat) dan ~38% (selamat).
*  **Keterbatasan/Pertanyaan: ** Preprocessing ini hanya membersihkan data mentah yang ada tanpa menggali informasi tambahan.

